registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspCsvServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+    private static final String ASPSP_NAME_COLUMN_NAME = "aspspName";
+    private static final ObjectReader CSV_OBJECT_READER;
+
+    private final AspspRepository aspspRepository;
+    private final AspspCsvRecordConverter csvRecordConverter;
+
+    static {
+        CSV_OBJECT_READER = new CsvMapper()
+                                    .readerWithTypedSchemaFor(AspspCsvRecord.class)
+                                    .withHandler(new DeserializationProblemHandler() {
+                                        @Override
+                                        public Object handleWeirdStringValue(DeserializationContext ctxt, Class<?> targetType, String valueToConvert, String failureMsg) {
+                                            if (targetType.isEnum()) {
+                                                return Enum.valueOf((Class<Enum>) targetType, valueToConvert.trim().toUpperCase());
+                                            }
+
+                                            return DeserializationProblemHandler.NOT_HANDLED;
+                                        }
+                                    });
+    }
+
+    public AspspCsvServiceImpl(AspspRepository aspspRepository, AspspCsvRecordConverter csvRecordConverter) {
+        this.aspspRepository = aspspRepository;
+        this.csvRecordConverter = csvRecordConverter;
+    }
+
+    @Override
+    public byte[] exportCsv() {
+        List<AspspPO> pos = aspspRepository.findAll();
+
+        return pos.stream()
+                       .map(csvRecordConverter::toAspspCsvRecord)
+                       .map(this::toCsvString)
+                       .collect(Collectors.joining())
+                       .getBytes();
+    }
+
+    @Override
+    public void importCsv(byte[] file) {
+        List<AspspPO> aspsps = readAllRecords(file);
+        aspspRepository.deleteAll();
+        aspspRepository.saveAll(aspsps);
+    }
+
+    private String toCsvString(AspspCsvRecord aspsp) {
+        CsvMapper mapper = new CsvMapper();
+        CsvSchema schema = mapper.schemaFor(AspspCsvRecord.class).withoutQuoteChar();
+
+        if (aspsp.getAspspName() == null) {
+            aspsp.setAspspName("");
+        }
+
+        if (aspsp.getAspspName().contains(",")) {
+            int nameColumnIndex = schema.column(ASPSP_NAME_COLUMN_NAME).getIndex();
+            schema = mapper.configure(CsvGenerator.Feature.STRICT_CHECK_FOR_QUOTING, true)
+                             .schemaFor(AspspCsvRecord.class)
+                             .rebuild()
+                             .replaceColumn(nameColumnIndex, new CsvSchema.Column(nameColumnIndex, ASPSP_NAME_COLUMN_NAME))
+                             .build();
+        }
+
+        try {
+            return mapper.writer(schema).writeValueAsString(aspsp);
+        } catch (JsonProcessingException e) {
+            logger.warn("Exception occurred while indexes were being written into a CSV: {}", e.getMessage());
```

Comment:
this is anti-pattern `log and re-throw`, just re-throw with given message

Reply:
my bad - was not that patient while copy-pasting code from the adapter app.

Reply:
Done, thanks